packages/reward-root-submitter/reward_root_submitter/main.py

```diff
@@ -1,58 +1,97 @@
 import logging
-import os
-import time
+import re
+from dataclasses import dataclass
 
-import schedule
-import sentry_sdk
-from dotenv import load_dotenv
+import pyarrow.parquet as pq
+from hexbytes import HexBytes
 from web3 import Web3
 
-from .submitter import RootSubmitter
-
-load_dotenv()  # take environment variables from .env
-for expected_env in [
-    "EVM_FULL_NODE_URL",
-    "OWNER",
-    "OWNER_PRIVATE_KEY",
-    "REWARD_POOL_ADDRESS",
-    "REWARD_PROGRAM_OUTPUT",
-    "ENVIRONMENT",
-]:
-    if expected_env not in os.environ:
-        raise ValueError(f"Missing environment variable {expected_env}")
-
-LOGLEVEL = os.environ.get("LOGLEVEL", "WARNING").upper()
-logging.basicConfig(level=LOGLEVEL)
-SENTRY_DSN = os.environ.get("SENTRY_DSN")
-if SENTRY_DSN is not None:
-    sentry_sdk.init(
-        SENTRY_DSN,
-        # Set traces_sample_rate to 1.0 to capture 100%
-        # of transactions for performance monitoring.
-        # We recommend adjusting this value in production.
-        traces_sample_rate=1.0,
-        environment=os.environ.get("ENVIRONMENT"),
+from .contracts import RewardPool
+
+NULL_HEX = HexBytes(
+    "0x0000000000000000000000000000000000000000000000000000000000000000"
+)
+
+EMPTY_MARKER_HEX = HexBytes(
+    "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+)
+
+
+@dataclass
+class MerkleRoot:
+    reward_program_id: str
+    payment_cycle: int
+    merkle_root_hash: HexBytes
+
+
+def setup_logging(config):
+    logging.basicConfig(level=config.log_level.upper())
+
+
+def safe_regex_group_search(regex, string, group):
+    """
+    Returns None in the case of a missing group
+    """
+    match = re.search(regex, string)
+    if match:
+        return match.group(group)
+    else:
+        return None
+
+
+def get_merkle_root_details(reward_output_filename):
+
+    reward_program_id = safe_regex_group_search(
+        r"rewardProgramID=([^/]*)", str(reward_output_filename), 1
     )
+    if not Web3().isChecksumAddress(reward_program_id):
+        raise Exception(
+            f"{reward_output_filename} does not have a valid checksummed address for the reward program ID"
+        )
+    payment_cycle = safe_regex_group_search(
+        r"paymentCycle=(\d*)", str(reward_output_filename), 1
+    )
+    if not (payment_cycle or "").isdigit():
```

Comment:
This particular if condition is confusing. 

```
payment_cycle = "1"
not (payment_cycle or "").isdigit()
```

yields false 

Is that meant to be?

Reply:
I think it's correct but awkward. I'll have a quick think if casting to an int and raising on an exception there with a better error message would change anything.

Reply:
Changed the regex so it requires at least one digit, and it's only pulling in digits so a "is not None" is enough.